Add profile_fields capability and deprecations into text section (#2)
tcpipuk/matrix-spec: content/client-server-api/_index.md

When this capability is not listed, clients should use `"1"` as the
default and only stable `available` room version.

### `m.profile_fields` capability

This capability has a flag, `enabled`, and two lists, `allowed` and
`disallowed`, that together denote which fields the user is able to
change via the profile endpoints.

When `enabled` is `false`, all profile fields are managed by the server
and the client is not permitted to make any changes.

When `enabled` is `true`, clients are permitted to modify profile fields,
subject to the restrictions implied by the OPTIONAL lists `allowed` and
`disallowed`. If only `allowed` is present, clients can modify all
contained fields but SHOULD assume all other fields to be managed by
the server. Contrarily, if only `disallowed` is present, clients are
unable to modify any contained fields but SHOULD assume all other fields
to be unmanaged. If both `allowed` and `disallowed` are specified,
`allowed` takes precendece. This means clients can modify all fields
in `allowed` but none of the fields in `disallowed` unless they also
occur in `allowed`. If neither `allowed` nor `disallowed` is present,

> clients can modify all fields without restrictions.

Author: This feels like quite a mouthful. 😕

I'm also still not sure if I'm grasping the intended evaluation logic correctly. Is that it?

1. If `allowed` is present:
   1. If `allowed` contains the field, allow.
   2. Otherwise, reject.
2. If `disallowed` is present and contains the field, reject.
3. Otherwise, allow.

CC @richvdh

Author: I thought OPTIONAL and MAY were interchangeable.

> MAY   This word, or the adjective "OPTIONAL", mean that an item is truly optional.

I don't really have a strong opinion on whether to use it in the spec though so `**optional**` seems fine to me as well.

When not listed, clients SHOULD assume the user is able to change
profile fields without any restrictions, provided the homeserver
advertises a specification version that includes the `m.profile_fields`
capability in the [`/versions`](/client-server-api/#get_matrixclientversions)
response.

An example of the capability API's response for this capability is:

```json
{
  "capabilities": {
    "m.profile_fields": {
      "enabled": true,
      "disallowed": ["displayname"]
    }
  }
}
```

### `m.set_displayname` capability

{{% boxes/note %}}
{{% changed-in v="1.16" %}}
This capability is now deprecated. Clients SHOULD use the
[`m.profile_fields`](/client-server-api/#mprofile_fields-capability)
capability instead.

For backwards compatibility, servers that directly or indirectly include
the `displayname` profile field in the `m.profile_fields` capability

> MUST still set this capability accordingly.

Author: Yeah, I meant "set this capability accordingly" in the sense of "do the needful so that it reflects whatever you did with `m.profile_fields`". You're variant is much clearer. 👍

> MUST still set this capability accordingly.

Author: Have fixed the `avatar_url` case with https://github.com/tcpipuk/matrix-spec/pull/2/files#r2256843044.

### `m.set_avatar_url` capability

{{% boxes/note %}}
{{% changed-in v="1.16" %}}
This capability is now deprecated. Clients SHOULD use the
[`m.profile_fields`](/client-server-api/#mprofile_fields-capability)
capability instead.

> For backwards compatibility, servers that directly or indirectly include
> the `avatar_url` profile field in the `m.profile_fields` capability
> MUST still set this capability accordingly.

Author: As per https://github.com/tcpipuk/matrix-spec/pull/2/files#r2256452750.

```suggestion
For backwards compatibility, servers that forbid setting the 
`avatar_url` profile field in the `m.profile_fields` capability
MUST still present this capability with `"enabled": false`.
```
